ambry-named-mysql/src/main/java/com/github/ambry/named/MySqlNamedBlobDb.java

```diff
@@ -256,16 +257,27 @@ public CompletableFuture<NamedBlobRecord> get(String accountName, String contain
         new GetTransactionStateTracker(remoteDatacenters, localDatacenter);
     return executeTransactionAsync(accountName, containerName, true, (accountId, containerId, connection) -> {
       long startTime = System.currentTimeMillis();
-      NamedBlobRecord record, recordV2;
-      record = run_get(accountName, containerName, blobName, option, accountId, containerId, connection);
+      NamedBlobRecord record = null, recordV2 = null;
+      try {
+        record = run_get(accountName, containerName, blobName, option, accountId, containerId, connection);
+      } catch (Exception eOld) {
+        if (!config.dbRelyOnNewtable) {
+          throw eOld;
+        }
+        logger.error("NAMED GET: Failed in pulling data from old table: ", eOld);
+      }
+
       if (config.dbTransition) {
         try {
           recordV2 = run_get_v2(accountName, containerName, blobName, option, accountId, containerId, connection);
-          if (!record.equals(recordV2)) {
+          if (record != null && !recordV2.equals(record)) {
             logger.warn("NAMED GET: Data Inconsistence: old record='{}', new record='{}'", record, recordV2);
             metricsRecoder.namedDataInconsistentGetCount.inc();
           }
         } catch (Exception e) {
+          if (config.dbRelyOnNewtable) {
+            throw e;
```

Comment:
Metric here too? We can use this to surface if any requests are failing due to new or old value.